tests/console/docker.pm

```diff
@@ -119,18 +120,19 @@ sub run {
         assert_script_run('man -P cat docker config | grep "docker-config - Manage Docker configs"');
     }
 
-    # Try to stop container using ctrl+c
-    my $sleep_time = 60 * get_var('TIMEOUT_SCALE', 1);
-    type_string("docker run --rm opensuse/tumbleweed sleep $sleep_time\n");
-    type_string("#Press ctrl+c ");
-    send_key 'ctrl-c';
-    type_string("#still in container\n");
-    # If echo works then ctrl-c stopped sleep
+    # Try to stop container with init script using ctrl+c
+    type_string("docker run --rm --init -it opensuse/tumbleweed sleep $sleep_time\n");
+    sleep 15; send_key 'ctrl-c'; sleep 15;    # Kill the container so it will not end successfully
```

Comment:
sleep's are never a good idea. Can you log the output to serial and wait_serial on it?

Reply:
Might not work - the container has to be connected to a PTY for Ctrl-C, not sure how that works together with output redirection.

Reply:
Easy to test out locally. I had no problems with the equivalent of `docker run --rm --init -it opensuse/tumbleweed sh -xc 'sleep 30' > /dev/$serialdev`

@pdostal I am however not sure what you are trying to achieve. I assume the problem you might have hit is that docker has not yet initialized the container fully but is still downloading or what? In any case *please* avoid sleep with magic numbers. This is in most cases wasting time and inviting for tricky to debug race conditions

Reply:
@okurz the problem is, I am testing the behavisor with / without the `--init` parameter, so I cannot use `sh -xc 'sleep 30'` because I need to execute just the `sleep` but nothing else.

Reply:
I see. So the "--init" makes a difference. However, still, sleep in code is quite dangerous and should be avoided

Reply:
@okurz yes, I agree, this is ugly as hell and unstable as ... anyways on line `116` there is a command utilizing the `--init` as well so I would like to remove this part. Do you agree?

Reply:
Well, it was you that added the `--init` part so I am completely fine to remove it again, your decision :)